From 75257283248c99347089225befee851e2d6e2895 Mon Sep 17 00:00:00 2001
From: Adrian <github@eskaan.de>
Date: Mon, 1 Jan 2024 07:55:18 +0100
Subject: [PATCH] Fix the 32 bit builds by changing a type cast (#1202)

`i64::from` was replaced by a direct `as` type cast.

Tested on i686-unknown-linux-gnu
---
 display-servers/xlib-display-server/src/xwrap/setters.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/display-servers/xlib-display-server/src/xwrap/setters.rs b/display-servers/xlib-display-server/src/xwrap/setters.rs
index f0cb569ed..efd4034b7 100644
--- a/display-servers/xlib-display-server/src/xwrap/setters.rs
+++ b/display-servers/xlib-display-server/src/xwrap/setters.rs
@@ -94,7 +94,7 @@ impl XWrap {
 
     /// Sets a desktop property.
     pub fn set_desktop_prop(&self, data: &[u32], atom: c_ulong) {
-        let x_data: Vec<c_long> = data.iter().map(|x| i64::from(*x)).collect();
+        let x_data: Vec<c_long> = data.iter().map(|x| *x as c_long).collect();
         self.replace_property_long(self.root, atom, xlib::XA_CARDINAL, &x_data);
     }
